### PR TITLE
Fix list jobs command argument parsing bug

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fixes issues introduced by #1416 .

### Description

- `nargs="*"` can't be combined with `const`
- add a simple unit test of arg parsing of list jobs

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: tests/unit_test/private/fed/server/job_cmds_test.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+from nvflare.private.fed.server.job_cmds import _create_list_job_cmd_parser
+
+TEST_CASES = [
+    (["-d", "-a", "12345", "-n", "hello_", "-m", "3"], Namespace(a=True, d=True, job_id="12345", m=3, n="hello_")),
+    (["12345", "-d", "-a", "-n", "hello_", "-m", "3"], Namespace(a=True, d=True, job_id="12345", m=3, n="hello_")),
+    (["-d", "-a", "-n", "hello_", "-m", "3"], Namespace(a=True, d=True, job_id=None, m=3, n="hello_")),
+    (["-a", "-n", "hello_", "-m", "3"], Namespace(a=True, d=False, job_id=None, m=3, n="hello_")),
+    (["-a"], Namespace(a=True, d=False, job_id=None, m=5, n=None)),
+    (["nvflare"], Namespace(a=False, d=False, job_id="nvflare", m=5, n=None)),
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply tests/unit_test/private/fed/server/job_cmds_test.py:28]: I already have tests without -m option.

File: nvflare/private/fed/server/job_cmds.py [hunk 2]
```diff
@@ -325,8 +329,7 @@ def list_jobs(self, conn: Connection, args: List[str]):
 
                 filtered_jobs.sort(key=lambda job: job.meta.get(JobMetaKey.SUBMIT_TIME.value, 0.0), reverse=True)
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_cmds.py:331]: Default is set to 5 which is the same as the previous code's behavior

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_cmds.py:331]: OK I will change the default behavior to be the same as 2.2

**Review by nvidianz — APPROVED** [review]

File: nvflare/private/fed/server/job_cmds.py [hunk 3]
```diff
@@ -56,6 +56,25 @@
...
+    parser = SafeArgumentParser(prog="list_jobs")
+    parser.add_argument("job_id", nargs="?", help="Job ID prefix")
+    parser.add_argument("-d", action="store_true", help="Show detailed list")
+    parser.add_argument(
+        "-a",
+        action="store_true",
+        help="List all jobs",
+    )
+    parser.add_argument("-n", help="Filter by job name prefix")
+    parser.add_argument(
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_cmds.py:65]: I think we just define [-m num_of_jobs] since this arg is newly introduced.

**Review by nvidianz — APPROVED** [review]

**Pull request merged**